src/redux/history/methods.ts

```diff
@@ -0,0 +1,198 @@
+  storeLocalHistory()
+}
+
+export const toggleHistoryEntryPinning = async (noteId: string): Promise<void> => {
+  const state = store.getState().history
+  const entryToUpdate = state.find(entry => entry.identifier === noteId)
+  if (!entryToUpdate) {
+    return Promise.reject(`History entry for note '${ noteId }' not found`)
+  }
+  if (entryToUpdate.pinStatus === undefined) {
+    entryToUpdate.pinStatus = false
+  }
+  entryToUpdate.pinStatus = !entryToUpdate.pinStatus
+  if (entryToUpdate.origin === HistoryEntryOrigin.LOCAL) {
+    updateLocalHistoryEntry(noteId, entryToUpdate)
+  } else {
+    const historyUpdateDto = historyEntryToHistoryEntryUpdateDto(entryToUpdate)
+    updateHistoryEntryRedux(noteId, entryToUpdate)
+    await updateHistoryEntryPinStatus(noteId, historyUpdateDto)
+  }
+}
+
+export const downloadHistory = (): void => {
+  const history = store.getState().history
+  history.forEach((entry: Partial<HistoryEntry>) => {
+    delete entry.origin
+  })
+  const json = JSON.stringify({
+    version: 2,
+    entries: history
+  } as HistoryExportJson)
+  download(json, `history_${ Date.now() }.json`, 'application/json')
+}
+
+export const mergeHistoryEntries = (a: HistoryEntry[], b: HistoryEntry[]): HistoryEntry[] => {
+  const noDuplicates = a.filter(entryA => !b.some(entryB => entryA.identifier === entryB.identifier))
+  return noDuplicates.concat(b)
+}
+
+export const convertV1History = (oldHistory: V1HistoryEntry[]): HistoryEntry[] => {
+  return oldHistory.map(entry => ({
+    identifier: entry.id,
+    title: entry.text,
+    tags: entry.tags,
+    lastVisited: DateTime.fromMillis(entry.time)
+                         .toISO(),
+    pinStatus: entry.pinned,
+    origin: HistoryEntryOrigin.LOCAL
+  }))
+}
+
+export const refreshHistoryState = async (): Promise<void> => {
+  const localEntries = loadLocalHistory()
+  if (!store.getState().user) {
+    setHistoryEntries(localEntries)
+    return
+  }
+  const remoteEntries = await loadRemoteHistory()
+  const allEntries = mergeHistoryEntries(localEntries, remoteEntries)
+  setHistoryEntries(allEntries)
+}
+
+export const storeLocalHistory = (): void => {
+  const history = store.getState().history
+  const localEntries = history.filter(entry => entry.origin === HistoryEntryOrigin.LOCAL)
+  const entriesWithoutOrigin = localEntries.map(entry => ({
+    ...entry,
+    origin: undefined
+  }))
+  window.localStorage.setItem('history', JSON.stringify(entriesWithoutOrigin))
+}
+
+export const storeRemoteHistory = (): Promise<void> => {
+  if (!store.getState().user) {
+    return Promise.resolve()
+  }
+  const history = store.getState().history
+  const remoteEntries = history.filter(entry => entry.origin === HistoryEntryOrigin.REMOTE)
+  const remoteEntryDtos = remoteEntries.map(historyEntryToHistoryEntryPutDto)
+  return postHistory(remoteEntryDtos)
+}
+
+const loadLocalHistory = (): HistoryEntry[] => {
+  const localJson = window.localStorage.getItem('history')
```

Comment:
:banana: I would move this to Line 162. (or at least after the if block)

Reply:
Moved this after the if block.